server/dals/expression-dal.js

```diff
@@ -144,19 +141,21 @@ module.exports = {
         return sequelize.query(buildExpressionQuery(req.query),
                 { type: sequelize.QueryTypes.SELECT })
                     .then(function (stdname) {
+                        const convertToJSON = {
+                            "DegradationRates" : convertProductionDegradationRateToJSON(stdname, "degradation_rate"),
+                            "ProductionRates" : convertProductionDegradationRateToJSON(stdname, "production_rate"),
+                            "ExpressionData" : convertExpressionToJSON(
+                                stdname,
+                                req.query.dataset,
+                                expressionTimepointsByDataset[req.query.dataset],
+                                req.query.genes.split(",")
+                            )
```

Comment:
Same about functions.

Reply:
I fixed this too. It only calls the function it needs